mlflow/models/resources.py

```diff
@@ -0,0 +1,145 @@
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from dataclasses import dataclass
+from enum import Enum
+from typing import List
+
+
+class ResourceType(Enum):
+    DATABRICKS_VECTOR_SEARCH_INDEX_NAME = "databricks_vector_search_index_name"
+    DATABRICKS_VECTOR_SEARCH_ENDPOINT_NAME = "databricks_vector_search_endpoint_name"
+    DATABRICKS_EMBEDDINGS_ENDPOINT_NAME = "databricks_embeddings_endpoint_name"
+    DATABRICKS_LLM_ENDPOINT_NAME = "databricks_llm_endpoint_name"
+    DATABRICKS_CHAT_ENDPOINT_NAME = "databricks_chat_endpoint_name"
+
+
+class Providers(Enum):
+    DATABRICKS = "databricks"
+
+
+@dataclass
+class Resource(ABC):
+    """
+    Base class for all resources to define the resources needed to serve a model.
+
+    Args:
+        type (ResourceType): The resource type.
+        target_uri (str): The target URI where these resources are hosted.
+    """
+
+    type: ResourceType
+    target_uri: Providers
```

Comment:
```suggestion
    target_uri: str
```

Reply:
@sunishsheth2009 Can we address this?

Reply:
Was not addressed previously because of this: https://github.com/mlflow/mlflow/pull/11804#discussion_r1578822544

Will address it now. :)